database/migrations/2018_11_30_154729_recovery_codes.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class RecoveryCodes extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('recovery_codes', function (Blueprint $table) {
+            $table->increments('id');
+            $table->unsignedInteger('account_id');
+            $table->unsignedInteger('user_id');
+            $table->string('recovery', 20);
```

Comment:
Are you sure about the `20` here? If the recovery code is configurable in the `auth` file, we might have code way longer than this. Either you put the default limit for a string, or you remove the code to be configurable (imho).

Reply:
OK I've removed the limit here.